lib/service/analysis-time-logger.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+var async = require('async');
+var debug = require('../util/debug')('analysis-time-logger');
+
+function AnalysisTimeLogger(logger, batchClient) {
+    this.logger = logger;
+    this.batchClient = batchClient;
+}
+
+module.exports = AnalysisTimeLogger;
+
+AnalysisTimeLogger.prototype.log = function (rows) {
+    var self = this;
+    var logElapsedTime = self._logNodeElapsedTime.bind(self);
+    var nodesToLog = getNodesToLogFromRows(rows);
+
+    async.map(nodesToLog, logElapsedTime, function (err) {
+        if (err) {
+            return self.logger.error(err);
+        }
+    });
+};
+
+function getNodesToLogFromRows(rows) {
```

Comment:
I would try to remove duplicated job IDs.

What about reducing the rows by job id => node ids?
```
{ job_id: [nodeid0, nodeid1, ...] }
```